### DISCUSSION TITLE
How do you fail an invocation?

### DISCUSSION THREAD

**Discussion by chx** [brefphp/bref#911] [Support] [answered]
I am writing an SQS handler in bref, it calls an API and I'd like to signal Lambda about failure if there's any so it can retry automatically.

Reading https://github.com/brefphp/bref/issues/642 it's not at all clear what should be done.

Would a `throw \Exception` without anything  catching it suffice?

**Comment by mnapoli** [comment, marked as answer]
Hey, yes that's exactly that: throw an exception in the handler and let it bubble up to Bref (which will bubble up to Lambda and then SQS).

**Reply by chx** [reply]
Thanks, that's great, how do I close this :) ? 

I reposted to https://stackoverflow.com/a/67065713/308851 as well.

**Reply by mnapoli** [reply]
I don't think you can close it, you marked the answer I think that's it 😅

**Reply by chx** [reply]
For the archives: https://github.com/brefphp/bref/blob/master/src/Runtime/LambdaRuntime.php speaks to this API https://docs.aws.amazon.com/lambda/latest/dg/runtimes-api.html 

Now it all makes sense!
